### PR TITLE
Hostname validation and sync start if proxy settings changes

### PR DESCRIPTION
**Pull request by wiggiBe**
(good first issue)
1. Issue: 
[Proxy Settings] Validate for empty hostname when manual proxy is selected #5885 

2. Issue:
Proxy change in Settings dialog should trigger sync #3233

### CODE REVIEW

**Comment by ckamm** [comment]
@wiggiBe This looks good!

1) Did you sign the contribution agreement already? Can I help with this?

2) I'm not sure `startScheduledSyncSoon` is the right choice here. Wouldn't you want to schedule all folders? But that happens automatically when an account becomes connected. So maybe the actual thing you want to trigger is an account-connectivity-check? (so you don't have to wait for 30s)

`AccountState::checkConnectivity()` would do that.

**Comment by wiggiBe** [comment]
Sorry, I didn't. I will sign it right away.

Oh, I see. I thought it would trigger a almost instant sync in all folders which are scheduled.
Do AccountState::checkConnectivity() really trigger a sync for a signed account?

**Comment by ckamm** [comment]
@wiggiBe In the use case of "The account is disconnected and the user enters the proxy information" it would do it - the account will connect and all folders will be scheduled. 

You also want to schedule all folders in cases where the account was already connected? (say, switching between two working proxys)

In that case you'd need to set the AccountState to Disconnected first. Then checkConnectivity would attempt using the new proxy and, if successful, set the state to Connected again and schedule all folders.

**Comment by wiggiBe** [comment]
That wasn't my intention but it makes sense to disconnect first and connect the account again.
I will do that.

File: src/gui/folderman.h [hunk 1]
```diff
@@ -277,9 +280,6 @@ private slots:
     /* unloads a folder object, does not delete it */
     void unloadFolder(Folder *);
 
-    /** Will start a sync after a bit of delay. */
-    void startScheduledSyncSoon();
```
Comment (guruz) [inline src/gui/folderman.h]: Why did you move this? :-)
Use ```git show``` before pushing to see the commit if it makes sense.
You can use ```git commit --amend``` to change your existing commit.

Comment (ckamm) [reply src/gui/folderman.h]: It was private and needed to be public. Should be undone when the Account reconnection is used instead.

Comment (guruz) [reply src/gui/folderman.h]: @ckamm Meh, github didn't display the context properly. I thought he changed it from "private **slots**" to not being a slot.

Comment (wiggiBe) [reply src/gui/folderman.h]: @ckamm Indeed I will undo that.

@guruz Thank you for you help. :)
Unfortunately git commit --amend dont work because the last commit I deleted some comments but I allready changed the postion.

**Comment by wiggiBe** [comment]
@ckamm Do I have to do a connectivity check for every saved account? (So I have to get a list of all accounts from accountmanager)

**Comment by ckamm** [comment]
@wiggiBe Yes, I'd iterate through the accounts, disconnect them (there's no function for that yet, you need to add one) and trigger the connectivity check.

File: src/gui/networksettings.cpp [hunk 2]
```diff
@@ -155,12 +156,25 @@ void NetworkSettings::saveProxySettings()
...
+        QString hostLineEditValue = _ui->hostLineEdit->text();
+
+        if(hostLineEditValue.isEmpty())
+        {
+            _ui->hostLineEdit->setStyleSheet("border: 1px solid red");
+            return;
+        }
+        else
+        {
+            _ui->hostLineEdit->setStyleSheet("border: 1px solid black"); // or blue
```
Comment (ogoffart) [inline src/gui/networksettings.cpp]: did you mean  "border: native" ?
or just `_ui->hostLineEdit->setStyleSheet(QString())`

Comment (wiggiBe) [reply src/gui/networksettings.cpp]: Yes, thank you. "border: native" is very good.

**Comment by ogoffart** [comment]
The change does not build:

Jenkins error:  

```
/var/lib/jenkins/workspace/cloud-client_client_PR-6140-VJ2XIDBENLXHDCMPYD3F44RPONFPGAIGK6ABLT3J66BPSU4OUQQQ/src/gui/networksettings.cpp: In member function ‘void OCC::NetworkSettings::saveProxySettings()’:
/var/lib/jenkins/workspace/cloud-client_client_PR-6140-VJ2XIDBENLXHDCMPYD3F44RPONFPGAIGK6ABLT3J66BPSU4OUQQQ/src/gui/networksettings.cpp:192:33: error: base operand of ‘->’ has non-pointer type ‘QList<QExplicitlySharedDataPointer<OCC::AccountState> >’
     for(int i=0; i < accountList->count(); i++)
                                 ^
/var/lib/jenkins/workspace/cloud-client_client_PR-6140-VJ2XIDBENLXHDCMPYD3F44RPONFPGAIGK6ABLT3J66BPSU4OUQQQ/src/gui/networksettings.cpp:196:13: error: ‘acouuntList’ was not declared in this scope
             acouuntList[i]->disconnectByProxySettings();
             ^

```

File: src/gui/networksettings.cpp [hunk 3]
```diff
@@ -185,8 +187,20 @@ void NetworkSettings::saveProxySettings()
     // start the sync.
     FolderMan::instance()->setDirtyProxy(true);
 
-    // start sync after a change
-    FolderMan::instance()->startScheduledSyncSoon();
+    QList<AccountStatePtr> accountList = AccountManager::instance()->accounts();
+
+    for(int i=0; i < accountList->count(); i++)
```
Comment (ckamm) [inline src/gui/networksettings.cpp]: `for (auto account : accountList) {`

**Comment by ogoffart** [comment]
@wiggiBe  are you still working on this?

**Comment by ckamm** [comment]
I've reworked and merged it to master!
